### DISCUSSION TITLE
Adding Loaders in routing of different pages

### DISCUSSION THREAD

**Discussion by Faizan711** [vercel/next.js#52136] [Help]
### Summary

Hello everybody, I am building a multi-page website using Nextjs and I am using Pages router for routing. I want to add a loader which will be shown before site goes to any page in website but I am having difficulties finding resources how to do it, If anyone knows how to do so then please help me with this. 
@shuding @delbaoliveira  Can you help me with this?

**Comment by icyJoseph** [comment]
You can follow a pattern similar to nprogress + next.js - since you are in the pages dir you can use Router events, to control showing the "loader" and hiding the "loader" as the SPA transitions between pages. 

- https://caspertheghost.me/blog/nprogress-next-js

The last bit would be to show the loader as the app renders, which you can do in the `_app` file,

```ts
import * as React from 'react';
import 'nprogress/nprogress.css';
import '../styles/globals.css';
import Router from 'next/router';
import NProgress from 'nprogress';

if (typeof window !== 'undefined') {
  NProgress.start();
}

export default function MyApp({ Component, pageProps }) {
  React.useEffect(() => {
    if (NProgress.isStarted()) {
      NProgress.done();
    }

    const handleRouteStart = () => NProgress.start();
    const handleRouteDone = () => NProgress.done();

    Router.events.on('routeChangeStart', handleRouteStart);
    Router.events.on('routeChangeComplete', handleRouteDone);
    Router.events.on('routeChangeError', handleRouteDone);

    return () => {
      // Make sure to remove the event handler on unmount!
      Router.events.off('routeChangeStart', handleRouteStart);
      Router.events.off('routeChangeComplete', handleRouteDone);
      Router.events.off('routeChangeError', handleRouteDone);
    };
  }, []);

  return <Component {...pageProps} />;
}
```

Live example: https://stackblitz.com/edit/nextjs-moqe3w?file=package.json,pages%2Findex.js,pages%2Fabout.js,pages%2F_app.js

**Reply by Faizan711** [reply]
Thanks for the information @icyJoseph , I will try this and let you know if it worked

**Comment by Faizan711** [comment]
Hey @icyJoseph, while implementing, I came across another library called NextNProgrss and it did the same functionality but it is much simple. You can check it out below, but thanks for your answer.
[https://www.npmjs.com/package/nextjs-progressbar](url)

**Reply by icyJoseph** [reply]
Yeah, I am not a big fan of installing libraries, it is already a high commitment for me to install nprogress itself, but of course you do as you prefer :)
